lib/ros_message.h

```diff
@@ -24,6 +37,11 @@ class RosMessage {
   {
   }
 
+  RosMessageRawBufferData raw_data() const {
```

Comment:
I believe the `raw_data_len` only gets set during hydration of data; we may want to call that here to ensure that its properly populated.

Reply:
Alternatively also if the raw_data_len calculation is trivial, we could move it to the constructor